src/sbml/SBMLErrorLog.cpp

```diff
@@ -385,7 +385,7 @@ SBMLErrorLog::getErrorWithSeverity(unsigned int n, unsigned int severity) const
   {
     if (matcher(*it))
     {
-      if (count == n) return dynamic_cast<const SBMLError*>(*it);
+      if (count == n) return static_cast<const SBMLError*>(*it);
```

Comment:
this cannot work, since the iterator could be pointing to an error of type `XMLError`, which is the base class of SBML error. So the dynamic cast was there to return a NULL error in that case. The static cast would just force the error object into an SBML error, and create a bad object. 

Thus the test failures: 

```

2021-10-01T19:34:00.9150270Z 35:  Generic Attribute Functions
2021-10-01T19:34:00.9655721Z 35: 99%: Checks: 1737, Failures: 1, Errors: 0
2021-10-01T19:34:00.9657744Z 35: D:\a\libsbml\libsbml\src\sbml\test\TestSBMLError.cpp:243:F:SBMLError:test_SBMLErrorLog_getErrorWithSeverity_XMLError:0: Assertion 'error->getErrorId() == XMLContentEmpty' failed
2021-10-01T19:34:00.9695739Z  35/184 Test  #35: test_sbml_sbml_run ............................***Failed    2.04 sec
2021-10-01T19:34:00.9700201Z test 36
```